src/main/kotlin/org/sol4k/VersionedTransaction.kt

```diff
@@ -0,0 +1,85 @@
+package org.sol4k
+
+import okio.Buffer
+import java.math.BigDecimal
+import java.math.RoundingMode
+import java.util.Base64
+import kotlin.math.max
+
+class VersionedTransaction(
+    val message: Message,
+    val signatures: MutableList<String>,
+) {
+
+    fun sign(keypair: Keypair) {
+        val data = message.serialize()
+        val signature = keypair.sign(data)
+        for (i in 0 until message.header.numRequireSignatures) {
+            val a = message.accounts[i]
+            if (a.verify(signature, data)) {
+                if (signatures.isEmpty()) {
+                    signatures.add(Base58.encode(signature))
+                } else {
+                    signatures[i] = Base58.encode(signature)
+                }
+                break
+            }
+        }
+    }
+
+    fun serialize(): ByteArray {
+        if (signatures.isEmpty() || signatures.size != message.header.numRequireSignatures) {
+            throw Exception("Signature verification failed")
```

Comment:
Let's introduce something like `SerializationException` inherited from `RuntimeException` and use it for serialization/deserialization purposes. Using `Exception` is too general in this case. Please check other places where it is applicable

Reply:
Indeed